### PR TITLE
[arm] Make the generic trampoline call mono_thread_force_interruption…

### PR DESCRIPTION
**Pull request by vargaz**
…_checkpoint_noraise () and throw the exception itself instead of calling the obsolete mono_interruption_checkpoint_from_trampoline ().

### CODE REVIEW

File: mono/mini/tramp-arm.c [hunk 1]
```diff
@@ -386,6 +388,50 @@ mono_arch_create_generic_trampoline (MonoTrampolineType tramp_type, MonoTrampInf
...
+	 * EH is initialized after trampolines, so get the address of the variable
+	 * which contains throw_exception, and load it from there.
+	 */
+	if (aot) {
+		/* Not really a jit icall */
+		code = mono_arm_emit_aotconst (&ji, code, buf, ARMREG_IP, MONO_PATCH_INFO_JIT_ICALL_ADDR, "throw_exception_addr");
+	} else {
+		ARM_LDR_IMM (code, ARMREG_IP, ARMREG_PC, 0);
+		ARM_B (code, 0);
+		*(gpointer*)code = mono_get_throw_exception_addr ();
```
Comment (lewurm) [inline mono/mini/tramp-arm.c]: why is `ARM_B (code, 0)` needed here? can't you use `-4` as offset for the `LDR_IMM` insn?

File: mono/mini/tramp-arm.c [hunk 2]
```diff
@@ -386,6 +388,50 @@ mono_arch_create_generic_trampoline (MonoTrampolineType tramp_type, MonoTrampInf
...
+		code += 4;
+	}
+	ARM_LDR_IMM (code, ARMREG_IP, ARMREG_IP, 0);
+	/* Branch to the throw trampoline */
+	/* lr contains the return address, the trampoline will use it as the throw site */
+	code = emit_bx (code, ARMREG_IP);
+	*(int*)code = 0xef9f0001;
+	code += 4;
```
Comment (lewurm) [inline mono/mini/tramp-arm.c]: what is this?

File: mono/mini/tramp-arm.c (hunk 1 above)
Comment (vargaz) [reply mono/mini/tramp-arm.c]: Copy pasted code.

File: mono/mini/tramp-arm.c (hunk 2 above)
Comment (vargaz) [reply mono/mini/tramp-arm.c]: Fixed.

**Pull request merged**